python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -188,6 +187,180 @@ def createAllocationFile(self, input):
+                unique_labels = set(job_labels)
+
+                #
+                # Make a jobs dictionary with the unique labels as keys
+                #
+                if verbose:
+                    print("Loop over unique job label list")
+                label_dict = {}
+                for job_label in unique_labels:
+                    hash = hashlib.sha1(job_label.encode("UTF-8")).hexdigest()
+                    shash = hash[:6]
+                    empty_list = []
+                    label_dict[job_label] = empty_list
+
+                # Loop over the Large jobs and categorize
+                for jid in list(condorq_bps_large.keys()):
+                    ajob = condorq_bps_large[jid]
+                    this_label = ajob["bps_job_label"]
+                    this_list = label_dict[this_label]
+                    this_list.append(ajob)
+
+                for job_label in unique_labels:
+                    if verbose:
+                        print(f"\n{job_label}")
+                    existingGlideinsIdle = 0
+                    numberOfGlideinsRed = 0
+                    numberOfGlideins = 0
+                    alist = label_dict[job_label]
+                    thisMemory = alist[0]["RequestMemory"]
+                    useCores = alist[0]["RequestCpus"]
+                    if useCores < autoCPUs:
+                        useCores = autoCPUs
+                    hash = hashlib.sha1(job_label.encode("UTF-8")).hexdigest()
+                    shash = hash[:6]
+                    numberOfGlideins = len(alist)
+                    jobname = f"{auser}_{shash}"
+                    print(f"{job_label} {jobname} target {numberOfGlideins}")
+
+                    # Do not submit squeue commands rapidly
+                    time.sleep(2)
+                    # Check Slurm queue Idle Glideins
+                    batcmd = f"squeue --noheader --states=PD --name={jobname} | wc -l"
+                    if verbose:
+                        print("The squeue command is: %s " % batcmd)
+                    try:
+                        resultPD = subprocess.check_output(batcmd, shell=True)
+                    except subprocess.CalledProcessError as e:
+                        print(e.output)
+                    existingGlideinsIdle = int(resultPD.decode("UTF-8"))
+                    if verbose:
+                        print(f"existingGlideinsIdle {jobname}")
+                        print(existingGlideinsIdle)
+
+                    numberOfGlideinsRed = numberOfGlideins - existingGlideinsIdle
+                    if verbose:
+                        print(f"{job_label} {jobname} reduced {numberOfGlideinsRed}")
+
+                    cpuopt = f"--cpus-per-task {useCores}"
+                    memopt = f"--mem {thisMemory}"
+                    jobopt = f"-J {jobname}"
+                    cmd = f"sbatch {cpuopt} {memopt} {jobopt} {generatedSlurmFile}"
+                    if verbose:
+                        print(cmd)
```

Comment:
Do we not have to include the number of existing large glideins that are running?

Reply:
The "number of existing large glideins that are running" is used in placing an upper bound on resource usage, and I was deferring that upper bound handling to later consideration.  Our usage of large jobs appears to be a very small number at the end of a workflow (they are gathering results together in some sense) and so i did not expect the large jobs to occur in a scenario of very high Slurm cluster usage.

Reply:
I was thinking of running glideins that had not yet matched idle jobs.  But that's harder to detect.  Should we have just some cap(s) to prevent a runaway case where jobs aren't matching and glideins aren't idling out fast enough?  Or is that not likely to happen?

Reply:
I am not worried yet about the rare coincidence case (glidein submitted earlier just starts to run and hasn't matched the idle job yet right when allocateNodes auto comes around for a next iteration.)

But some cap on the number of "cores worth of glideins" being submitted, yes that would certainly be prudent.  Basically a  10000 core footprint say should not be exceeded  maybe 5000 for Large glidein routine.   We'll have to see if that is feasible in this branch.